book/getting-started/run-a-wallaroo-application-docker.md

```diff
@@ -0,0 +1,201 @@
+
+- `-v /tmp/wallaroo-docker/python-virtualenv:/src/python-virtualenv`: This flag will mount a host directory as a data volume within the container. The first time this is run for the provided directory, this command will setup a persistent Python Virtual Environment for the container on your host. Thus, if you need to install any python modules using pip or easy_install they will persist after the container is stopped or deleted. This flag is optional, but without it, you will not have a persistent Python Virtual Environment for the container.
+
+- `--name wally`: The name for the container. This flag is optional but makes it easier to reference the container when we open other Bash sessions in the container.
+
+## Terminal 2, Start the Metrics UI
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+This command will start a new Bash shell within the container, we enter via the `environment-setup.sh` script to ensure we're in our persistent Python Virtual Environment.
+
+To start the Metrics UI run:
+
+```bash
+metrics_reporter_ui start
+```
+
+You can verify it started up correctly by visiting [http://localhost:4000](http://localhost:4000).
+
+If you need to restart the UI, run:
+
+```bash
+metrics_reporter_ui restart
+```
+
+When it's time to stop the UI, run:
+
+```bash
+metrics_reporter_ui stop
+```
+
+If you need to start the UI after stopping it, run:
+
+```bash
+metrics_reporter_ui start
+```
+
+## Terminal 3, Run Giles Receiver
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+We'll use Giles Receiver to listen for data from our Wallaroo application.
+
+```bash
+receiver --listen 127.0.0.1:5555 --no-write --ponythreads=1 --ponynoblock
+```
+
+You should see the `Listening for data` that indicates that Giles receiver is running.
+
+## Terminal 4, Run the "Celsius to Fahrenheit" Application
+
+Enter the Wallaroo Docker container:
+
+```bash
+docker exec -it wally environment-setup.sh
+```
+
+First, we'll need to get to the python Celsius example directory with the following command:
+
+```bash
+cd /src/wallaroo/examples/python/celsius
+```
+
+Now that we are in the proper directory, and the metrics UI and a data receiver are up and running, we can run the application itself by executing the following command:
+
+```bash
+machida --application-module celsius --in 127.0.0.1:7000 \
+  --out 127.0.0.1:5555 --metrics 127.0.0.1:5001 --control 127.0.0.1:6000 \
+  --data 127.0.0.1:6001 --name worker-name --external 127.0.0.1:5050 \
+  --cluster-initializer --ponythreads=1 --ponynoblock
+```
+
+This tells the "Celsius to Fahrenheit" application that it should listen on port `7000` for incoming data, write outgoing data to port `5555`, and send metrics data to port `5001`.
```

Comment:
"This tells" -> 

This command tells machida to:
* run the Celsius to Fahrenheit application
* listen on port 7000 for incoming data
* send output to port 5555
* send metrics data to port 5001